docs/aws/audit/elbmonitoring/rules/elb_no_deletion_policy_remediation.mdx

```diff
@@ -1,6 +1,206 @@
+   - Click "Save" to apply the changes.
+
+By following these steps, you can ensure that your ELBs have the Deletion Protection flag enabled, preventing accidental deletions.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent Elastic Load Balancers (ELBs) from being accidentally deleted by enabling the deletion protection flag using AWS CLI, follow these steps:
+
+1. **Identify the Load Balancer:**
+   First, you need to identify the ARN (Amazon Resource Name) of the load balancer you want to protect. You can list all your load balancers using the following command:
+   ```sh
+   aws elbv2 describe-load-balancers
+   ```
+
+2. **Enable Deletion Protection:**
+   Use the `modify-load-balancer-attributes` command to enable deletion protection for the specified load balancer. Replace `<load-balancer-arn>` with the ARN of your load balancer.
+   ```sh
+   aws elbv2 modify-load-balancer-attributes --load-balancer-arn <load-balancer-arn> --attributes Key=deletion_protection.enabled,Value=true
+   ```
+
+3. **Verify the Configuration:**
+   To ensure that the deletion protection flag has been enabled, describe the load balancer attributes using the following command:
+   ```sh
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+   Look for the `deletion_protection.enabled` attribute in the output and ensure its value is `true`.
+
+4. **Automate for Multiple Load Balancers (Optional):**
+   If you have multiple load balancers and want to enable deletion protection for all of them, you can use a loop in a shell script. Here is an example:
+   ```sh
+   for lb_arn in $(aws elbv2 describe-load-balancers --query 'LoadBalancers[*].LoadBalancerArn' --output text); do
+       aws elbv2 modify-load-balancer-attributes --load-balancer-arn $lb_arn --attributes Key=deletion_protection.enabled,Value=true
+   done
+   ```
+
+By following these steps, you can ensure that deletion protection is enabled for your Elastic Load Balancers, preventing accidental deletions.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration where Elastic Load Balancers (ELBs) should have the Deletion Protection flag enabled in AWS, you can use the Boto3 library in Python. Here are the steps to achieve this:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables.
+
+3. **Enable Deletion Protection for Existing ELBs**:
+   Use the following Python script to enable deletion protection for all existing ELBs in your AWS account.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>